### PR TITLE
Add pipe boolean option to transport.consume()

### PR DESCRIPTION
**Pull request by ibc**
- So the receiver will get all streams.
- It works for any kind of transport (but PipeTransport which is always like this by design).
- Replaces PR #476.
- **TODO:** Document it.

### CODE REVIEW

File: src/Transport.ts [hunk 1]
```diff
@@ -543,26 +544,29 @@ export class Transport extends EnhancedEventEmitter
...
+			}
 		}
 
 		const internal = { ...this._internal, consumerId: uuidv4(), producerId };
 		const reqData =
 		{
 			kind                   : producer.kind,
 			rtpParameters,
-			type                   : producer.type,
+			type                   : !pipe ? producer.type : 'pipe',
```
Comment (jmillan) [inline src/Transport.ts]: cosmetic, just for readability:

`type: pipe ? pipe : producer.type`

**Review by jmillan — APPROVED** [review]
:+1:

File: src/Transport.ts (hunk 1 above)
Comment (ibc) [reply src/Transport.ts]: done

**Comment by nazar-pc** [comment]
Does this mean that `PipeTransport` is now similar to `PlainTransport` with all consumers having `pipe: true` or there are some other differences?

File: src/ortc.ts [hunk 2]
```diff
@@ -1043,53 +1044,74 @@ export function getConsumerRtpParameters(
...
 
-	// Set a single encoding for the Consumer.
-	consumerParams.encodings!.push(consumerEncoding);
+			if (rtxSupported)
+				encoding.rtx = { ssrc: baseRtxSsrc + i };
+			else
+				delete encoding.rtx;
 
-	// Copy verbatim.
-	consumerParams.rtcp = consumableParams.rtcp;
```
Comment (nazar-pc) [inline src/ortc.ts:1092]: Was RTCP removed intentionally here?

Comment (ibc) [reply src/ortc.ts:1092]: It was already copied in the object created above at the beginning of the method.

**Comment by ibc** [comment]
> Does this mean that `PipeTransport` is now similar to `PlainTransport` with all consumers having `pipe: true` or there are some other differences?

Right, differences are:

- plain transport still needs receiver rtpCapabilities in its consume() API.
- plain transport allows TCC and REMB

In v4 we are gonna merge both somehow.

**Comment by nazar-pc** [comment]
I already have single `consume()` implementation in Rust port instead of separate for PipeTransport and with growing number of `if` statements it is harder to manage (updating it to match latest `v3` changes right now).

**Comment by ibc** [comment]
That's why we want to merge both in v4 and simplify things. They have ended being the same thing with artificial constraints in both.

File: src/ortc.ts (hunk 2 above)
Comment (albyho) [reply src/ortc.ts:1092]: Got it.

**Pull request merged**